### PR TITLE
[nlohmann-json] Always install multiple headers to allow forward declarations

### LINKED ISSUE #10795: [nlohmann-json] Support multiple header install
****

**Is your feature request related to a problem? Please describe.**
When using JSON for Modern C++ (nlohmann-json), there are cases when a developer would rather forward-declare (say, in a header) rather than include the entire `json.hpp`. The vcpkg port currently just downloads the single-include header and does not provide other options.

**Proposed solution**
According to [upstream documentation](https://github.com/nlohmann/json#integration)
```
You can further use file include/nlohmann/json_fwd.hpp for forward-declarations. The installation of json_fwd.hpp (as part of cmake's install step), can be achieved by setting -DJSON_MultipleHeaders=ON.
```
so by cloning and building via CMake, vcpkg can provide both the core and multiheader features.

I'd like to try and do this. I assume it involves replacing the nlohmann-json portfile with a more standard CMake library portfile?

### PR DESCRIPTION
**Pull request by kevinlul**
Closes #10795

Should work everywhere that the single include already does.

- [x] Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

**Comment by ras0219** [comment]
Is there a reason to not simply enable this always?

File: ports/nlohmann-json/portfile.cmake [hunk 1]
```diff
@@ -18,12 +18,30 @@ download_src(LICENSE.MIT d5f7bb6a33469e19250a5e20db44e7ba09602ee85bc0afb03e49324
...
+    set(FEATURE_OPTIONS -DJSON_MultipleHeaders=ON)
+else()
+    download_src(single_include/nlohmann/json.hpp 6643748f7d96196e7b5659f39762a839e509da3352787ced29d62c28be18dd74b53bc2601cae3c765117ba3cb23aed491de9766ea66dd2539894f77073d0398d)
+endif()
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
-    OPTIONS -DJSON_BuildTests=0
+    OPTIONS -DJSON_BuildTests=0 ${FEATURE_OPTIONS}
```
Comment (NancyLi1013) [inline ports/nlohmann-json/portfile.cmake:44]:
```suggestion
    OPTIONS 
        -DJSON_BuildTests=0 
        ${FEATURE_OPTIONS}
```

File: ports/nlohmann-json/portfile.cmake [hunk 2]
```diff
@@ -18,12 +18,30 @@ download_src(LICENSE.MIT d5f7bb6a33469e19250a5e20db44e7ba09602ee85bc0afb03e49324
...
+    )
+    vcpkg_extract_source_archive_ex(
+        OUT_SOURCE_PATH OUT_PATH
+        ARCHIVE ${ARCHIVE}
+        REF ${SOURCE_VERSION}
+        WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR}
+        NO_REMOVE_ONE_LEVEL
+    )
+    file(COPY "${OUT_PATH}/include" DESTINATION ${SOURCE_PATH})
+    set(FEATURE_OPTIONS -DJSON_MultipleHeaders=ON)
```
Comment (NancyLi1013) [inline ports/nlohmann-json/portfile.cmake:36]: Could you please use `vcpkg_check_features()` to handle the feature?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @kevinlul 
Thanks for this PR.
Have you tested the feature `multiple-headers` and if it can work on at least  x86-windows, x64-windows, x64-windows-static and x64-linux triplets?

File: ports/nlohmann-json/portfile.cmake (hunk 2 above)
Comment (kevinlul) [reply ports/nlohmann-json/portfile.cmake:36]: Is there a point to doing this when there's already a branch for the different download? `if ("multiple-headers" IN_LIST FEATURES)`

**Comment by kevinlul** [comment]
Yes, I have checked on those triplets, made easier by the fact that this is a header-only library. However, I noticed that since #11941 and the 3.8.0 update, because the library now ships its own CMake module files, two directories are created in `installed/TRIPLET/share`: `nlohmann-json` for the vcpkg copyright since that's the port name here, and `nlohmann_json` for the CMakes because that's the self-ascribed name of the library.

File: ports/nlohmann-json/portfile.cmake (hunk 2 above)
Comment (NancyLi1013) [reply ports/nlohmann-json/portfile.cmake:36]: Sorry, I ignored this point before. The current status is okay.

**Review by NancyLi1013 — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**